linux/mcompile/keymap/mc_import_rules.cpp

```diff
@@ -53,51 +52,53 @@ void DeadKey::KMX_AddDeadKeyRow(KMX_WCHAR baseCharacter, KMX_WCHAR combinedChara
 
 bool DeadKey::KMX_ContainsBaseCharacter(KMX_WCHAR baseCharacter) {
   std::vector<KMX_WCHAR>::iterator it;
-  for(it=this->m_rgbasechar.begin(); it<m_rgbasechar.end(); it++) {
-    if(*it == baseCharacter) {
+  for (it = this->m_rgbasechar.begin(); it < m_rgbasechar.end(); it++) {
+    if (*it == baseCharacter) {
       return true;
     }
   }
   return false;
 }
 
-int KMX_ToUnicodeEx(guint keycode, PKMX_WCHAR pwszBuff, int shift_state_pos, int caps, GdkKeymap *keymap) {
-  GdkKeymapKey *maps;
-  guint *keyvals;
+int KMX_ToUnicodeEx(guint keycode, PKMX_WCHAR pwszBuff, int shift_state_pos, int caps, GdkKeymap* keymap) {
+  GdkKeymapKey* maps;
+  guint* keyvals;
   gint count;
 
   if (!gdk_keymap_get_entries_for_keycode(keymap, keycode, &maps, &keyvals, &count))
     return 0;
 
-  if (!(ensureValidInputForKeyboardTranslation( shift_state_pos, count, keycode)))
+  if (!(ensureValidInputForKeyboardTranslation(shift_state_pos, count, keycode)))
     return 0;
 
-  KMX_DWORD keyVal= (KMX_DWORD) KMX_get_KeyVal_From_KeyCode(keymap, keycode, ShiftState(shift_state_pos), caps);
+  KMX_DWORD keyVal = (KMX_DWORD)KMX_get_KeyVal_From_KeyCode(keymap, keycode, ShiftState(shift_state_pos), caps);
   std::u16string str = convert_DeadkeyValues_To_U16str(keyVal);
-  pwszBuff[0]= * (PKMX_WCHAR)  str.c_str();
-
+  pwszBuff[0] = *(PKMX_WCHAR)str.c_str();
 
   g_free(keyvals);
   g_free(maps);
 
-  if((keyVal >=  deadkey_min) && (keyVal <=  deadkey_max))          // deadkeys
+  if (u16len(pwszBuff) < 1)
+    return 0;
+
+  if ((keyVal >= deadkey_min) && (keyVal <= deadkey_max))   // deadkeys
     return -1;
-  else if(gdk_keyval_to_unicode(keyVal)  == 0)                      // NO UNICODE
+  else if (gdk_keyval_to_unicode(keyVal) == 0)              // NO UNICODE
     return 0;
-  else                                                              // usable char
+  else                                                      // usable char
     return 1;
 }
 
-int KMX_DeadKeyMap(int index, std::vector<DeadKey *> *deadkeys, int deadkeyBase, std::vector<KMX_DeadkeyMapping> *deadkeyMappings) {   // I4327   // I4353
-  for(size_t i = 0; i < deadkeyMappings->size(); i++) {
-    if((*deadkeyMappings)[i].deadkey == index) {
+KMX_WCHAR KMX_DeadKeyMap( int index, std::vector<DeadKey*>* deadkeys, int deadkeyBase, std::vector<KMX_DeadkeyMapping>* deadkeyMappings) {  // I4327   // I4353
```

Comment:
It might be a bit easier to pass the vectors as references, but I might have already mentioned that elsewhere :smile: 

```suggestion
KMX_WCHAR KMX_DeadKeyMap(int index, std::vector<DeadKey*>& deadkeys, int deadkeyBase, std::vector<KMX_DeadkeyMapping>& deadkeyMappings) {  // I4327   // I4353
```

Reply:
on windows it`s done like that -> similarity issue #11802